Generate test resources with make all

rth opened the pull request:
This aims to improve the user workflow for running tests. Running pytest directly has multiple advantages when compared to `make test` (including the abilty to run a subset of tests, using other plugins etc). At the same time, currently there is a risk of not generating the needed resources files that have to be created with,
```
make build/test.html build/test_data.txt
```

When those are missing, for any reason, the error message is unhelpful, and can lead to long debugging time as in https://github.com/iodide-project/pyodide/pull/127#issuecomment-416345852

This PR,
 - improves the error message if `build/tests.html` is missing for any reason (there is [no way to get the HTTP response status with selenium](https://stackoverflow.com/a/6512785/1791279))
 - auto-copies `build/tests.html` and `build/test_data.txt` at each pytest session with a session level fixture. This only copies ~540 bytes, so I don't think it's an issue to copy them each time. The advantages is that after `make`  one can directly run pytest with the desired options.
 - factorizes some pytest options in `setup.cfg` to avoid copying them each time in CI etc.

rth commented:
This PR also includes a proper fix for https://github.com/iodide-project/pyodide/pull/130 where the issue was that,
 - locally we can indeed use `clang-format` from emsdk (and this is what should be happening)
 - in Circle CI we mount the cash with emsdk (and run make) after linting  which means that we need to use the system `clang-format`. Besides  the cached `esmsdk` will not exist if its `Makefile` was modified.

mdboom commented on .circleci/config.yml:
```diff
@@ -93,7 +92,7 @@ jobs:
 
             source pyodide-env/bin/activate
             export PATH=$PWD/firefox:$PATH
-            pytest test -v --instafail -k firefox
```
Why was this removed?  This is really useful if a test times out on Circle-CI -- if we don't do this, we don't get any output from the test (or any other failed tests).

mdboom replied:
Never mind.  I see you've added to the config below.

rth commented:
Thanks for the review! I see your point about not modifying the `build/` folder in tests.

Putting it under `make all` would also work -- addressed your comments.

mdboom commented on test/conftest.py:
```diff
@@ -39,6 +39,10 @@ def __init__(self):
+        if not (BUILD_PATH / 'test.html').exists():
+            # selenium does not expose HTTP response codes
+            raise ValueError(f"{(BUILD_PATH / 'test.html').resolve()} "
+                             f"does not exist!")
```
I think this needs to be removed also?

mdboom replied:
Never mind.  I see this is just the warning.  :+1:

rth replied:
Yes, it just aims to address the fact that when `wait.until(PyodideInited())` fails we end up with an unhelpful error message and it takes `N_tests*timeout ~ 2 hours` to complete. This detects one of reasons this may fail, and error early in that case.
